core/src/main/scala/kafka/server/KafkaApis.scala

```diff
@@ -969,6 +963,53 @@ class KafkaApis(val requestChannel: RequestChannel,
     }
   }
 
+  def handleCreateTopicRequest(request: RequestChannel.Request) {
+    val createTopicRequest = request.body.asInstanceOf[CreateTopicRequest]
+
+    val results =
+      if (!authorize(request.session, Create, Resource.ClusterResource)) {
+        createTopicRequest.topics.asScala.map { case (topic, _) =>
+          (topic, Errors.CLUSTER_AUTHORIZATION_FAILED)
+        }
+      }
+      else {
+        val brokers = metadataCache.getAliveBrokers.map(_.id)
+        createTopicRequest.topics.asScala.map { case (topic, arguments) =>
+          try {
+            val configs = new Properties()
+            arguments.configs.asScala.foreach { case (key, value) =>
+              configs.setProperty(key, value)
+            }
+            LogConfig.validate(configs)
+
+            val assignments = {
+              if (!arguments.replicasAssignments.isEmpty) {
+                // Note: we don't check that replicaAssignment doesn't contain unknown brokers - unlike in add-partitions case,
+                // this follows the existing logic in TopicCommand
+                arguments.replicasAssignments.asScala.map { case (partitionId, replicas) =>
```

Comment:
`AdminUtils.createOrUpdateTopicPartitionAssignmentPathInZK(...)`, called below, has the validations. I didn't want to duplicate them if possible. I will make sure all validations that are needed are there.